### ISSUE TITLE
s2Member Pro-Forms

### ISSUE THREAD

**Issue by jaswsinc** [websharks/s2member-kb#120]
---
##### KB Article Creation Checklist

- [x] Write initial draft for this KB Article; label this issue `draft` and either `questions` or `tutorials`
- [x] Add [required YAML configuration](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration)
- [x] Add Tags for this KB Article to the YAML config (see [YAML Keys (Explained)](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration#yaml-keys-explained))
- [x] Edit and finalize draft for publishing (remove `draft` label, add `draft-finalized` label)
- [x] Assign Issue to yourself and [create Markdown file](https://github.com/websharks/team/wiki/KB-Articles-::-Creating-the-Markdown-File) (remove `draft-finalized` label, add `pending`)
- [x] Project Lead: Review and Publish KB Article (remove `pending` label, add `published` label)

##### Additional TODOs

_Additional TODO list items go here._

---

**Markdown File:** http://wsharks.com/1910qOW
**Published KB Article:** http://s2member.com/kb-article/s2member-pro-forms/

---

**Comment by ryanaoneil** [comment]
How do you add s2member pro to composer? Didn't see that listed in the knowledge base.

**Comment by jaswsinc** [comment]
@ryanaoneil Sorry for the delayed response here.

Are you referring to Composer https://getcomposer.org/ ?

**Comment by ryanaoneil** [comment]
yes

On Wed, Jun 3, 2015 at 4:52 PM, JasWSInc <notifications@github.com> wrote:

> @ryanaoneil <https://github.com/ryanaoneil> Sorry for the delayed
> response here.
>
> Are you referring to Composer https://getcomposer.org/ ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/websharks/s2member-kb/issues/120#issuecomment-108626374>
> .
>



-- 
Ryan O'Neil
LOGOSIdentity.com
We get the Word out.

**Comment by jaswsinc** [comment]
Copy that. TY. The s2Member repo just needs a `composer.json` file so that it could be included as a Composer dependency. I'll open an issue for this here: https://github.com/websharks/s2member/issues/591

**Comment by ryanaoneil** [comment]
Sweet! What about the pro version? Currently I'm pulling in s2member via
wpackagist.

**Comment by jaswsinc** [comment]
It won't be possible for the pro version I'm afraid, because we don't publish that on GitHub publicly, for obvious reasons :-) i.e., it is a pro copy available only via s2Member.com at this time.

**Comment by ryanaoneil** [comment]
Yeah, I get that on the pro version. Other pro versions have private repos.

Here's an example:

        {
            "type": "package",
            "package": {
                "name": "deliciousbrains/wp-migrate-db-pro",
                "type": "wordpress-plugin",
                "version": "1.4.6",
                "dist": {
                    "type": "zip",
                    "url": "
https://deliciousbrains.com/dl/wp-migrate-db-pro-latest.zip?licence_key=
<license_key>&site_url=composer-test.dev"
                },
                "require": {
                    "composer/installers": "v1.0.7"
                }
            }
        },

On Wed, Jun 3, 2015 at 5:29 PM, JasWSInc <notifications@github.com> wrote:

> It won't be possible for the pro version I'm afraid, because we don't
> publish that on GitHub publicly, for obvious reasons :-) i.e., it is a pro
> copy available only via s2Member.com at this time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/websharks/s2member-kb/issues/120#issuecomment-108632516>
> .
>



-- 
Ryan O'Neil
LOGOSIdentity.com
We get the Word out.

**Comment by jaswsinc** [comment]
That's an interesting idea. I just discussed this (and some other options) with an associate of mine. I'll see what we can come up with. It may not happen right away, but we will definitely keep it in mind! Thank you for the tip there.

**Comment by ryanaoneil** [comment]
Yeah, thanks so much! I'd love to promote s2 more but that's a desperate
need. I just purchased the unlimited pro license because I believe in the
product enough. Appreciate the work ya'll do!

**Comment by raamdev** [comment]
@ryanaoneil Woohoo! Thanks so much for the support. :-)

**Comment by bizactuator** [comment]
Is there a kb article that will explain where I can put custom code after payment with stripe pro form that will 1) validate payment and 2) update a custom database with the purchase information?

I don't see where to put the code to verify payment and thank you code to update the database.
